alembic/versions/0021_added_created_and_last_modified_columns_.py

```diff
@@ -0,0 +1,372 @@
+"""
+Added created and last modified columns for all entities
+
+Revision ID: 0021
+Revises: 0020
+Create Date: 2023-11-22 16:05:12.088610
+
+"""
+import sqlalchemy as sa
+from alembic_utils.pg_function import PGFunction
+from alembic_utils.pg_trigger import PGTrigger
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "0021"
```

Comment:
One thought, that I'm not sure about but wanted to mention. That sometimes it would be tempting to separate out larger migrations like this into per-table changes. This has the downside of extra clutter as now there are more files, but makes individual ones smaller so easier to read and also means a smoother rollback, as if one of them had a problem you wouldnt have to roll all of them back. I don't have strong views on this so no pressure. What do you think?

Reply:
I've gone back through and separated things out a little more. It's still a bit janky, but considerably better!